File: restdoc-http-client/src/main/java/restdoc/client/config/RestDocProperties.java
package restdoc.client.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "restdoc")
public class RestDocProperties {

    private String serverIp;

    private int serverPort;

    private String service;

    public String getServerIp() {
        return serverIp;
    }

    public void setServerIp(String serverIp) {
        this.serverIp = serverIp;
    }

    public int getServerPort() {
        return serverPort;
    }

    public void setServerPort(int serverPort) {
        this.serverPort = serverPort;
    }

    public String getService() {
        return service;
    }

    public void setService(String service) {
        this.service = service;
    }
}

File: restdoc-rpc-client/restdoc-restweb-client/src/main/java/restdoc/client/config/RestDocProperties.kt
package restdoc.client.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConfigurationProperties(prefix = "restdoc")
@ConstructorBinding
@Deprecated(message = "")
data class RestDocProperties(
        val serverIp: String,
        val serverPort: Int,
        val service: String? = null)